index.js

```diff
@@ -188,15 +208,15 @@ FacebookInsightStream.prototype._collect = function ( metrics, item, buffer, eve
 
     // for the audience API, we just use one metric ['app_event']
     // with a few events
-    var _metric = metrics.shift() || options.metrics[ 0 ];
+    var _metric = metrics[ metrics.length -1 ] || options.metrics[ 0 ];
     var model = { id: item.id, metric: _metric }
 
     var _ev;
     var _agg;
     if ( hasEvents ) {
         // extend the query model with event name
         // and aggregation type
-        _ev = events.shift();
+        _ev = events[ events.length - 1 ];
```

Comment:
@oshribin Ok. Maybe it's worth adding a comment that we're not yet shifting/popping the item from it's list since it may be needed to be recollected again, if the case is indeed retryable .